e2e/system_test.go

```diff
@@ -0,0 +1,201 @@
+package e2e
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/minimal/crypto"
+	"github.com/0xPolygon/minimal/e2e/framework"
+	"github.com/0xPolygon/minimal/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/umbracle/go-web3"
+)
+
+type addressKeyPair struct {
+	privateKey *ecdsa.PrivateKey
+	address    types.Address
+}
+
+// generateAddressKeyPairs is a helper method for generating signing keys and addresses
+func generateAddressKeyPairs(num int, t *testing.T) []*addressKeyPair {
+	var pairs []*addressKeyPair
+
+	for i := 0; i < num; i++ {
+		senderKey, senderAddr := framework.GenerateKeyAndAddr(t)
+		pairs = append(pairs, &addressKeyPair{address: senderAddr, privateKey: senderKey})
+	}
+
+	return pairs
+}
+
+func TestSystem_StakeAmount(t *testing.T) {
+	addressKeyPairs := generateAddressKeyPairs(2, t)
+
+	signer := crypto.NewEIP155Signer(100)
+
+	validAccounts := []struct {
+		address types.Address
+		balance *big.Int
+	}{
+		// Valid account #1
+		{
+			addressKeyPairs[0].address,
+			framework.EthToWei(50), // 50 ETH
+		},
+		// Empty account
+		{
+			addressKeyPairs[1].address,
+			big.NewInt(0),
+		},
+	}
+
+	stakingAddress := types.StringToAddress("1001")
+
+	testTable := []struct {
+		name          string
+		staker        types.Address
+		stakeAmount   *big.Int
+		shouldSucceed bool
+	}{
+		{
+			"Valid stake",
+			validAccounts[0].address,
+			framework.EthToWei(10),
+			true,
+		},
+		{
+			"Invalid stake",
+			validAccounts[1].address,
+			framework.EthToWei(100),
+			false,
+		},
+	}
+
+	srvs := framework.NewTestServers(t, 1, func(config *framework.TestServerConfig) {
+		config.SetConsensus(framework.ConsensusDev)
+		config.SetSeal(true)
+
+		for _, acc := range validAccounts {
+			config.Premine(acc.address, acc.balance)
+		}
+	})
+	srv := srvs[0]
+
+	rpcClient := srv.JSONRPC()
+	for indx, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Fetch the staker balance before sending the transaction
+			accountBalance, err := rpcClient.Eth().GetBalance(
+				web3.Address(testCase.staker),
+				web3.Latest,
+			)
+			assert.NoError(t, err)
+
+			var out string
+			if callErr := rpcClient.Call(
+				"stake_getStakedBalance",
+				&out,
+				web3.Address(testCase.staker),
+				web3.Latest.String(),
+			); callErr != nil && testCase.shouldSucceed {
```

Comment:
It should return the result (0) even if the account doesn't exist, so no need for the second part of the condition, right?
```suggestion
			); callErr != nil {
```

Reply:
Yes. The test should fail if this call fails, regardless of the test case, since getStakedBalance returns 0 for unknown accounts, as you said.

Fixed 👍